### PR TITLE
Add the external module for SAML SSO

### PR DESCRIPTION
**Pull request by antepusic**
### Description

This PR adds an external module that enables SAML single sign-on by validating SAML responses. The supported identity providers are Microsoft Entra ID and Okta.

NB: Auth stack extensions have been made in #1990.

[master < Task] PR
- [x] Write E2E tests
- [x] Provide the full content or a guide for the final git message
    - Add the external module for SAML single sign-on

### CI Testing Labels
Please select the appropriate CI test labels _(CI -build=**build-name** -test=**test-suite**)_

### Documentation checklist
- [x] Add the documentation label tag
- [x] Add the bug / feature label tag
- [x] Add the milestone for which this feature is intended
- [x] Write a release note, including added/changed clauses
    - Memgraph now has enterprise support for single sign-on with Microsoft Entra ID and Okta over the SAML protocol.
 - [x] Link the documentation PR here
    - https://github.com/memgraph/documentation/pull/602#issuecomment-2137195293
- [x] Tag someone from docs team in the comments

### CODE REVIEW

File: src/auth/reference_modules/saml.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    auth = OneLogin_Saml2_Auth(request_data, saml_settings)
+
+    # Process response
+    try:
+        auth.process_response()
+    except Exception as e:
+        return {"authenticated": False, "errors": f"Errors while processing SAML response: {str(e)}"}
+    errors = auth.get_errors()
+    if errors:
+        return {"authenticated": False, "errors": f"Errors while processing SAML response: {'n'.join(errors)}"}
```
Comment (imilinovic) [inline src/auth/reference_modules/saml.py:137]: I think you mean't '\n'.join()

File: src/auth/reference_modules/saml.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+    return attributes[username_attribute]
+
+
+def get_role_mappings(raw_role_mappings: str):
+    if not raw_role_mappings:
+        return {}
+
+    role_mappings = "".join(raw_role_mappings.split(" "))  # remove whitespace
+    return {mapping.split(":")[0]: mapping.split(":")[1] for mapping in role_mappings.split(";")}
```
Comment (imilinovic) [inline src/auth/reference_modules/saml.py:88]: I think it's maybe better to have a path to json file that contains role mappings since it can be big

**Review by imilinovic — CHANGES_REQUESTED** [review]

File: src/auth/reference_modules/saml.py (hunk 2 above)
Comment (antepusic) [reply src/auth/reference_modules/saml.py:88]: Considered this; think we’re fine for now:
* the current way is consistent with the `--auth-module-mappings` flag, the only other SSO-related mapping

**Review by imilinovic — APPROVED** [review]
Looks good now but I am not 100% familiar with saml standards.

File: src/auth/reference_modules/saml.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/python3
+import io
+import json
+import os
+
+from onelogin.saml2.auth import OneLogin_Saml2_Auth
```
Comment (gitbuda) [inline src/auth/reference_modules/saml.py:6]: Add `requirements.txt` under `src/auth/reference_modules/` + install it during memgraph install, update documentation, how users should install all deps required to run reference modules.

**Comment by kgolubic** [comment]
@antepusic I'm missing the Release note entry and documentation PR.

**Comment by antepusic** [comment]
@kgolubic Added; it might be worth combining this PR’s release note with https://github.com/memgraph/memgraph/pull/2045 because they complement each other

File: src/auth/reference_modules/saml.py (hunk 3 above)
Comment (antepusic) [reply src/auth/reference_modules/saml.py:6]: Added

**Pull request merged**